text/0000-debuggable-macro-expansions.md

```diff
@@ -0,0 +1,219 @@
+- Feature Name: debuggable-macro-expansions
+- Start Date: 2017-08-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+By default, annotate code expanded from a macro with debug location info
+corresponding to the macro definition (i.e. the behavior that's currently
+available on nightly via `-Zdebug-macros`). Add an annotation
+`#[collapse_debuginfo]` to enable a particular macro definition to opt to have
+the expansion annotated with debug location info corresponding to the macro
+invocation (i.e. the current default behavior).
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the debug location info associated with code expanded from a macro
+claims that the location of the expanded code is the location of the macro
+invocation. This serves two purposes: First, it makes panic stacks point to
+the location in the code where a panicking assertion was invoked as opposed to
+pointing to the panic machinery internals. Second, it results in debuggers
+skipping over print-like macros like `println!` that programmers virtually
+always want to skip over when debugging and that due to current technical
+limitations cannot be made skippable without claiming collapsed location info
+for their expansion.
+
+However, the approach of collapsing debug location information yields bad
+results for macros that are not assert-like and not print-like but that expand
+to substantial code. The current approach of collapsing debug information by
+default makes non-assert-like, non-print-like macro usage result in code that
+is on debuggable, that doesn't get useful panic/crash stacks in CI or in the
```

Comment:
I think "on debuggable" should be changed "not debuggable" here, and "at that" should be changed to "and that" on the subsequent line.